starter-code/src/functions-and-arrays.js

```diff
@@ -40,6 +141,30 @@ const wordsUnique = [
   'bring'
 ];
 
+
+
+function uniquifyArray(wordsUnique) {
+  const wordsNotRepeated = [];
+
+  if (wordsUnique.length == 0) {
+    return [];
+  } else {
+    for (i = 0; i < wordsUnique.length; i++) {
+      z = i + 1;
+      if (wordsUnique.indexOf(wordsUnique[i], z) == -1) {
+
+        wordsNotRepeated.push(wordsUnique[i]);
+
+      }
+    }
+
+    return wordsNotRepeated;
+  }
+  console.log(wordsNotRepeated);
+}
+
```

Comment:
Esta función para mi es la más jodida de todo el ejercicio, bastante bien está ya lo que habéis conseguido. El tema está en que jasmine es un poco lerda y si pusheas a un nuevo array en lugar de borrar los elementos del antiguo se lia, porque no le quedan en el mismo orden.

De todas formas la opción de pushear a uno nuevo es mucho mejor así que no os preocupeis, todo ok!